### DISCUSSION TITLE
JNI/JVM bindings in mobile apps

### DISCUSSION THREAD

**Discussion by Demiguise** [MobileNativeFoundation/discussions#179] [General]
Hey all, am curious to hear how others are managing their JVM bindings with C++ code for mobile apps. Do you have your own frameworks for handling it? Some off-the-shelf solution? 

I know it can be difficult to sometimes keep track of the JNI asm notation, and it's prone to errors. Always annoying to forget a semicolon in an object classpath, compile and not fully figure out why there's no matching function for that signature :D

**Comment by mkattima1** [comment]
@Demiguise , I have been there while bridging hardware backed security (Trusted Execution Environment) with Android apps via SDK. Even if there exists a tool to generate boiler plate code, it's a good idea to verify argument type mappings. This becomes even more important when argument data flows across hardware boundaries.
